src/main/java/sg/ncl/MainController.java

```diff
@@ -1795,13 +1797,13 @@ public String startExperiment(
 
         if (!checkPermissionRealizeExperiment(realization, session)) {
             log.warn("Permission denied to start experiment: {} for team: {}", realization.getExperimentName(), teamName);
-            redirectAttributes.addFlashAttribute("message", "Permission deined. If the error persists, please contact support@ncl.sg");
+            redirectAttributes.addFlashAttribute("message", "Permission deined. If the error persists, please contact " + contactEmail);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Permission deined. If the error persists, please contact " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)